File: VB/DisableVisualStylesModule.Win/DisableVisualStylesWindowController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Skins
Imports DevExpress.XtraBars
Imports DevExpress.ExpressApp
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Controls
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.Win.Templates
Imports DevExpress.ExpressApp.Win.SystemModule
Imports DevExpress.ExpressApp.Win.Templates.ActionContainers

Namespace DisableVisualStylesModule.Win
	Public Class DisableVisualStylesWindowController
		Inherits WindowController
		Public Sub New()
			TargetWindowType = WindowType.Main
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			DisableVisualStyles()
		End Sub
		Protected Overrides Sub OnDeactivated()
			MyBase.OnDeactivated()
			RemoveHandler Application.CustomizeTemplate, AddressOf Application_CustomizeTemplate
		End Sub
		Private Sub Application_CustomizeTemplate(ByVal sender As Object, ByVal e As CustomizeTemplateEventArgs)
			ResolveTemplate(e.Template)
		End Sub
		Protected Overridable Sub DisableVisualStyles()
			If System.Windows.Forms.SystemInformation.TerminalServerSession Then
				AddHandler Application.CustomizeTemplate, AddressOf Application_CustomizeTemplate
				InitGlobalOptions()
			End If
		End Sub
		Protected Overridable Sub InitGlobalOptions()
			Animator.AllowFadeAnimation = False
			SkinManager.DisableFormSkins()
			SkinManager.DisableMdiFormSkins()
			BarAndDockingController.Default.PropertiesBar.MenuAnimationType = AnimationType.None
			BarAndDockingController.Default.PropertiesBar.SubmenuHasShadow = False
			BarAndDockingController.Default.PropertiesBar.AllowLinkLighting = False
			System.Windows.Forms.Application.VisualStyleState = System.Windows.Forms.VisualStyles.VisualStyleState.NoneEnabled
		End Sub
		Private Sub ResolveTemplate(ByVal template As IFrameTemplate)
			Dim formTemplate As XtraFormTemplateBase = TryCast(template, XtraFormTemplateBase)
			If formTemplate IsNot Nothing Then
				If formTemplate.FormStyle = RibbonFormStyle.Ribbon Then
					AddHandler formTemplate.RibbonTransformer.Transformed, AddressOf RibbonTransformer_Transformed
				Else
					InitBarOptions(formTemplate.BarManager)
					UserLookAndFeel.Default.SetWindowsXPStyle()
				End If
			End If
		End Sub
		Private Sub RibbonTransformer_Transformed(ByVal sender As Object, ByVal e As EventArgs)
			InitRibbonOptions((CType(sender, ClassicToRibbonTransformer)).Ribbon)
		End Sub
		Protected Overridable Sub InitRibbonOptions(ByVal ribbon As RibbonControl)
			If ribbon IsNot Nothing Then
				ribbon.ItemAnimationLength = 0
				ribbon.GroupAnimationLength = 0
				ribbon.PageAnimationLength = 0
				ribbon.ApplicationButtonAnimationLength = 0
				ribbon.GalleryAnimationLength = 0
				ribbon.TransparentEditors = False
				InitBarOptions(ribbon.Manager)
			End If
		End Sub
		Protected Overridable Sub InitBarOptions(ByVal manager As BarManager)
			If manager IsNot Nothing Then
				manager.AllowItemAnimatedHighlighting = False
			End If
		End Sub
	End Class
End Namespace

File: CS/DisableVisualStylesModule.Win/DisableVisualStylesWindowController.cs
using System;
using DevExpress.Skins;
using DevExpress.XtraBars;
using DevExpress.ExpressApp;
using DevExpress.LookAndFeel;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Controls;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.Win.Templates;
using DevExpress.ExpressApp.Win.SystemModule;
using DevExpress.ExpressApp.Win.Templates.ActionContainers;

namespace DisableVisualStylesModule.Win {
    public class DisableVisualStylesWindowController : WindowController {
        public DisableVisualStylesWindowController() {
            TargetWindowType = WindowType.Main;
        }
        protected override void OnActivated() {
            base.OnActivated();
            DisableVisualStyles();
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            Application.CustomizeTemplate -= Application_CustomizeTemplate;
        }
        private void Application_CustomizeTemplate(object sender, CustomizeTemplateEventArgs e) {
            ResolveTemplate(e.Template);
        }
        protected virtual void DisableVisualStyles() {
            if(System.Windows.Forms.SystemInformation.TerminalServerSession) {
                Application.CustomizeTemplate += Application_CustomizeTemplate;
                InitGlobalOptions();
            }
        }
        protected virtual void InitGlobalOptions() {
            Animator.AllowFadeAnimation = false;
            SkinManager.DisableFormSkins();
            SkinManager.DisableMdiFormSkins();
            BarAndDockingController.Default.PropertiesBar.MenuAnimationType = AnimationType.None;
            BarAndDockingController.Default.PropertiesBar.SubmenuHasShadow = false;
            BarAndDockingController.Default.PropertiesBar.AllowLinkLighting = false;
            System.Windows.Forms.Application.VisualStyleState = System.Windows.Forms.VisualStyles.VisualStyleState.NoneEnabled;
        }
        private void ResolveTemplate(IFrameTemplate template) {
            XtraFormTemplateBase formTemplate = template as XtraFormTemplateBase;
            if(formTemplate != null) {
                if(formTemplate.FormStyle == RibbonFormStyle.Ribbon) {
                    formTemplate.RibbonTransformer.Transformed += RibbonTransformer_Transformed;
                }
                else {
                    InitBarOptions(formTemplate.BarManager);
                    UserLookAndFeel.Default.SetWindowsXPStyle();
                }
            }
        }
        private void RibbonTransformer_Transformed(object sender, EventArgs e) {
            InitRibbonOptions(((ClassicToRibbonTransformer)sender).Ribbon);
        }
        protected virtual void InitRibbonOptions(RibbonControl ribbon) {
            if(ribbon != null) {
                ribbon.ItemAnimationLength = 0;
                ribbon.GroupAnimationLength = 0;
                ribbon.PageAnimationLength = 0;
                ribbon.ApplicationButtonAnimationLength = 0;
                ribbon.GalleryAnimationLength = 0;
                ribbon.TransparentEditors = false;
                InitBarOptions(ribbon.Manager);
            }
        }
        protected virtual void InitBarOptions(BarManager manager) {
            if(manager != null) {
                manager.AllowItemAnimatedHighlighting = false;
            }
        }
    }
}